File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # if x is negative return false because negative number is not
        # a palindrom
        # Add the number into a array then calculate test whether it is palindrom
        if x < 0:
            return False 

        arr = []
        while x > 0:
            arr.append(x % 10)
            x = x // 10

        i, j = 0, len(arr) - 1
        while i < j:
            if arr[i] == arr[j]:
                i += 1
                j -= 1
            else:
                return False

        return True

File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0) return false;

        long x_copy = x;
        long x_reversed = 0;

        while(x > 0) {
            x_reversed = (x_reversed * 10) + (x % 10);
            x = floor(x/10);
        }

        return x_copy == x_reversed;
    }
};